### PR TITLE
Update tests for changes to reporting of vector_size attribute

### PR DESCRIPTION
**Pull request by matt-gretton-dann**

### CODE REVIEW

**Comment by matt-gretton-dann** [comment]
Updated branch to enable QL Submodule merge downstream

**Comment by matt-gretton-dann** [comment]
Internal PR: https://git.semmle.com/Semmle/code/pull/34455

File: cpp/ql/test/library-tests/vector_sizes/code.c [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// semmle-extractor-options: --clang --edg --trap_container=folder --edg --trap-compression=none
```
Comment (ian-semmle) [inline cpp/ql/test/library-tests/vector_sizes/code.c]: I don't think we want the `--trap*` flags in the test. Note that you can `export SEMMLE_TRAP_CONTAINER=folder` to override it on the commandline.

File: cpp/ql/test/library-tests/ir/ir/raw_ir.expected [hunk 2]
```diff
@@ -5277,8 +5277,8 @@ ir.cpp:
 # 1153|     mu0_2(unknown)                                     = UnmodeledDefinition              : 
 # 1153|     r0_3(glval<int>)                                   = VariableAddress[i]               : 
 # 1153|     mu0_4(int)                                         = InitializeParameter[i]           : &:r0_3
-# 1154|     r0_5(glval<__attribute((vector_size(16))) int>)    = VariableAddress[vi4]             : 
-# 1154|     mu0_6(__attribute((vector_size(16))) int)          = Uninitialized[vi4]               : &:r0_5
+# 1154|     r0_5(glval<__attribute((vector_size(16UL))) int>)  = VariableAddress[vi4]             :
```
Comment (dbartol) [inline cpp/ql/test/library-tests/ir/ir/raw_ir.expected]: These types were all returned from `Type.getUnspecifiedType()`. Does that mean that all unspecified types for vector types will always print the size with the `UL` suffix, regardless of how they were declared?

I'm fine with the change in the string. I just want to make sure there's not some underlying issue with `getUnspecifiedType()` that's causing the change.

Comment (matt-gretton-dann) [reply cpp/ql/test/library-tests/ir/ir/raw_ir.expected]: See the library-tests/vector_sizes qltest I added in this patch for the behaviour.  It depends on how the `vector_size ()`attribute was declared.  Using `sizeof()` (which is common) will make the type `unsigned long` (in my environment)

File: cpp/ql/test/library-tests/vector_sizes/code.c (hunk 1 above)
Comment (matt-gretton-dann) [reply cpp/ql/test/library-tests/vector_sizes/code.c]: Pushed new changed which drop the --trap options

**Pull request merged**